visualization/react-lib/src/components/table/Table.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from "react";
+import {
+  DetailsList,
+  IColumn,
+  IDetailsList,
+  IGroup,
+  IDetailsGroupRenderProps,
+  SelectionMode
+} from "@fluentui/react";
+import { getTheme, mergeStyleSets } from "@fluentui/react/lib/Styling";
+
+const ROW_HEIGHT: number = 42; // from DEFAULT_ROW_HEIGHTS in DetailsRow.styles.ts
+const GROUP_HEADER_AND_FOOTER_SPACING: number = 8;
+const GROUP_HEADER_AND_FOOTER_BORDER_WIDTH: number = 1;
+const GROUP_HEADER_HEIGHT: number = 95;
+const GROUP_FOOTER_HEIGHT: number =
+  GROUP_HEADER_AND_FOOTER_SPACING * 4 +
+  GROUP_HEADER_AND_FOOTER_BORDER_WIDTH * 2;
+
+const theme = getTheme();
+
+const classNames = mergeStyleSets({
+  headerAndFooter: {
+    borderTop: `${GROUP_HEADER_AND_FOOTER_BORDER_WIDTH}px solid ${theme.palette.neutralQuaternary}`,
+    borderBottom: `${GROUP_HEADER_AND_FOOTER_BORDER_WIDTH}px solid ${theme.palette.neutralQuaternary}`,
+    padding: GROUP_HEADER_AND_FOOTER_SPACING,
+    margin: `${GROUP_HEADER_AND_FOOTER_SPACING}px 0`,
+    background: theme.palette.neutralLighterAlt,
+    // Overlay the sizer bars
+    position: "relative",
+    zIndex: 100,
+  },
+  headerTitle: [
+    theme.fonts.large,
+    {
+      padding: "4px 0",
+    },
+  ],
+});
+
+export interface IItem {
+  name: string;
+  value: string;
+  description: string;
+}
+
+export interface IState {
+  items: IItem[];
+  groups: IGroup[];
+  showItemIndexInView: boolean;
+  isCompactMode: boolean;
+}
+
+export class TableComponent extends React.Component<{ state: IState, columns: IColumn[] }, IState> {
+  private _root = React.createRef<IDetailsList>();
+  private _columns: IColumn[];
+
+  constructor(props: { state: IState, columns: IColumn[] }) {
+    super(props);
+
+    this.state = props.state;
+    this._columns = props.columns;
+  }
+
+  public componentWillUnmount() {
+    if (this.state.showItemIndexInView) {
+      const itemIndexInView = this._root.current!.getStartItemIndexInView();
+      alert("first item index that was in view: " + itemIndexInView);
+    }
+  }
+
+  public render() {
+    const { items, groups } = this.state;
+
+    return (
+      <div>
+        <DetailsList
```

Comment:
It feels that the second column with the values could be narrow. Most of the values are cut off, while at the same time the first column has a lot of empty space.

Reply:
That's intended. We try to give more space to 1st column to avoid it being overflow. The problem is there are tooltips on the 1st column, if overflow, it's very hard to show tooltip in side a hover text.

Reply:
Although, I did shorten the names quite a bit on the last pr Peter, so maybe we could give the other column a little more space now?

Reply:
Because I shortened the names, I gave a little more space to the number column. I tested with minimized window & 500% zoom and the text columns still didn't overflow :)